jina/executors/indexers/vector.py

```diff
@@ -495,7 +496,7 @@ def query(
             _query_vectors = _ext_A(_norm(vectors))
             dist = self._cosine(_query_vectors, self.query_handler)
         else:
-            raise NotImplementedError(f'{self.metric} is not implemented')
+            raise MetricTypeError(f'{self.metric} is not implemented')
```

Comment:
This line actually cannot be reached I think. Maybe need to refactor this part.

Reply:
Done